arkouda/array_view.py

```diff
@@ -0,0 +1,249 @@
+            raise ValueError(f"cannot reshape array of size {base.size} into shape {self.shape}")
+        self.base = base
+        self.size = base.size
+        self.dtype = base.dtype
+        self.ndim = self.shape.size
+        self.itemsize = self.base.itemsize
+        if order.upper() in {'C', 'ROW_MAJOR'}:
+            self.order = OrderType.ROW_MAJOR
+        elif order.upper() in {'F', 'COLUMN_MAJOR'}:
+            self.order = OrderType.COLUMN_MAJOR
+        else:
+            raise ValueError(f"cannot traverse with order={order}")
+        # cache _reverse_shape which is reversed if we're row_major
+        self._reverse_shape = self.shape if self.order is OrderType.COLUMN_MAJOR else self.shape[::-1]
+        if self.shape.min() == 0:
+            # avoid divide by 0 if any of the dimensions are 0
+            self._dim_prod = zeros(self.shape.size, self.dtype)
+        else:
+            # cache dim_prod to avoid recalculation, reverse if row_major
+            self._dim_prod = cumprod(self.shape)//self.shape if self.order is OrderType.COLUMN_MAJOR else cumprod(self._reverse_shape)//self._reverse_shape
+
+    def __len__(self):
+        return self.size
+
+    def __repr__(self):
+        from arkouda.client import pdarrayIterThresh
+        if self.size <= pdarrayIterThresh:
+            return self.to_ndarray().__repr__()
+        else:
+            vals = [f"'{self.base[i]}'" for i in range(3)]
+            vals.append('... ')
+            vals.extend([f"'{self.base[i]}'" for i in range(self.size-3, self.size)])
+        return f"array([{', '.join(vals)}]), shape {self.shape}"
+
+    def __str__(self):
+        from arkouda.client import pdarrayIterThresh
+        if self.size <= pdarrayIterThresh:
+            return self.to_ndarray().__str__()
+        else:
+            vals = [f"'{self.base[i]}'" for i in range(3)]
+            vals.append('... ')
+            vals.extend([f"'{self.base[i]}'" for i in range(self.size-3, self.size)])
+        return f"[{', '.join(vals)}], shape {self.shape}"
+
+    def __getitem__(self, key):
+        if isinstance(key, int) or isinstance(key, slice):
+            key = [key]
+        elif isinstance(key, tuple):
+            key = list(key)
+        if len(key) > self.ndim:
+            raise IndexError(f"too many indices for array: array is {self.ndim}-dimensional, but {len(key)} were indexed")
+        if len(key) < self.ndim:
+            # append self.ndim-len(key) many ':'s to fill in the missing dimensions
+            for i in range(self.ndim - len(key)):
+                key.append(slice(None, None, None))
+        try:
+            # attempt to convert to a pdarray (allows for view[0,2,1] instead of view[ak.array([0,2,1])]
+            # but pass on RuntimeError to allow things like view[0,:,[True,False,True]] to be correctly handled
+            key = array(key)
+        except:  # bare except is bad practice but we want to pass on any errors/warnings
+            pass
```

Comment:
This just seems like there should be a better way to handle this. Maybe at the very least, if we are only expecting a `RuntimeError`, on catch that in the except? That would at least error on any unexpected errors. 

I think a helper function here may be warranted to essentially handle these cases and check what key contains. If key is only numerics, we can create the pdarray. I think a helper function to make the decision here rather than a `try except` would be better suited.

Reply:
a helper function might be a good idea. I wasn't only getting a runtime error, there was also a numpy deprecation warning that i was trying to avoid from the `np.array` done within `ak.array`
```python
>>> np.array([0,1,[True,False,True]])
/home//miniconda3/envs/ak-work/bin/ipython:1: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
  #!/home//miniconda3/envs/ak-work/bin/python
array([0, 1, list([True, False, True])], dtype=object)

>>> In [9]: ak.array([0,1,[True,False,True]])
/home//proj/arkouda/arkouda/pdarraycreation.py:178: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray
  a = np.array(a)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-9-329ab0e78122> in <module>
----> 1 ak.array([0,1,[True,False,True]])

~/proj/arkouda/arkouda/pdarraycreation.py in array(a, dtype)
    208     # If not strings, then check that dtype is supported in arkouda
    209     if a.dtype.name not in DTypes:
--> 210         raise RuntimeError("Unhandled dtype {}".format(a.dtype))
    211     # Do not allow arrays that are too large
    212     size = a.size

RuntimeError: Unhandled dtype object
```

So even catching the `RuntimeError` and handling it correctly would still result in a distracting deprecation. I couldn't think of a better way to do it than this but it's far from ideal, so I'm open to suggestions

Reply:
@pierce314159 - I think this is fine for now. Can we add an issue to look into a more robust solution and work it when we have time? I don't think we need to hold the PR up for this right now.

Reply:
Issue #1333 created to track this